### ISSUE TITLE
How to query Left Join where the key of the 2nd table is null?

### ISSUE THREAD

**Issue by ujjwalguptaofficial** [ujjwalguptaofficial/JsStore#306]
### Discussed in https://github.com/ujjwalguptaofficial/JsStore/discussions/305

<div type='discussions-op-text'>

<sup>Originally posted by **alisabrigok** December  6, 2022</sup>
Basically, this is what I'm trying to query:

<img width="346" alt="Screenshot 2022-12-06 at 01 29 19" src="https://user-images.githubusercontent.com/6465827/205756654-a7b6c705-f31c-498f-bb7f-c21636bfd24a.png">

In IDBStudio, if I perform the below query;

The where clause doesn't return the records where the customerId of "Orders" is null. 

```
select({
    from: 'Customers',
    join: {
        with: 'Orders',
        on: "Orders.customerId=Customers.customerId",
        type: "left",
        as: {
            customerId: "orders.customerId"
        },
        where: {
            customerId: "null"
        },
    }
});
```

It just returns the whole "Customers" records with the columns of the "Orders" merged as all null.
I would expect this query to return me the records of "Customers" who don't have any "Orders".

Just to give a working SQL example, the below query on this website https://www.programiz.com/sql/online-compiler/ returns the exact desired result.

```
SELECT *
FROM Customers
LEFT JOIN Orders
ON Customers.customer_id = Orders.customer_id
WHERE Orders.customer_id IS Null
```

Since we can perform Left Join using JsStore as well, how to achieve this behavior? @ujjwalguptaofficial

Thanks in advance.</div>

**Comment by ujjwalguptaofficial** [comment]
fixed and published in v - 4.4.8

**Comment by ujjwalguptaofficial** [comment]
From version - 4.5.7, join column can be used in root where. the where in join will work like `and query` - 

e.g - 

```
select({
            from: 'Customers',
            join: {
                with: 'Orders',
                on: "Orders.customerId=Customers.customerId",
                type: "left",
                as: {
                    customerId: "orders_customerId"
                }
            },
            where: {
                orders_customerId: -1234
            }
        })
```
